### PR TITLE
create the calculateDriverUsage function

### PR DESCRIPTION
**Pull request by JingeXC**

### CODE REVIEW

File: src/data/routing/Routing.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+@description
+Calculate the usage % of driver.
+
+*/
+
+export var calculateDriverUsage = function(data) {
+
+    if (!data) {
+        throw 'data is empty';
```
Comment (chrishiste) [inline src/data/routing/Routing.js]: - Please use `new` keyword when throwing an error (https://google.github.io/styleguide/jsguide.html#features-exceptions)
- You should use a `try...catch` block whenever you use the `throw` keyword

File: src/data/routing/Routing.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Output
+percentage: string
+
+@description
+Calculate the usage % of driver.
+
+*/
+
+export var calculateDriverUsage = function(data) {
```
Comment (chrishiste) [inline src/data/routing/Routing.js]: - Please use an arrow function here:
```js
export const calculateDriverUsage = (data) => {...}
```
- Never use `var` it is not block scoped

File: src/data/routing/Routing.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    if (!data.driver_schedules || !data.driver_schedules.length) {
+        throw 'driver_schedules is empty';
+    }
+    if (!data.driver_assignments || !data.driver_assignments.length) {
+        throw 'driver_assignments is empty';
+    }
+
+    let driverSchedulesTime = null;
+    for (let schedules of data.driver_schedules) {
+        let startTime = new Date(schedules.start_at + " " + schedules.start_time).getTime();
```
Comment (chrishiste) [inline src/data/routing/Routing.js]: Consider using template string:
```js
new Date(`${schedules.start_at} ${schedules.start_time}`).getTime();
```

**Review by chrishiste — COMMENTED** [review]
Please take a look at my reviews and fix the problems

Also could you try to use cleaner commit message (https://github.com/RomuloOliveira/commit-messages-guide)

File: src/data/routing/Routing.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+Output
+percentage: string
+
+@description
+Calculate the usage % of driver.
+
+*/
+
+export const calculateDriverUsage = (data) => {
+    class exception {
```
Comment (ccwukong) [inline src/data/routing/Routing.js]: Please move this outside the arrow function

**Pull request merged**